### PR TITLE
Modular UI - initial version

### LINKED ISSUE #23: New Tab Page: Modular UI
Relates to https://github.com/ghostery/roadmap/issues/20

### PR DESCRIPTION
**Pull request by smalluban**
Fixes #23.

* [x] Structure 
* [x] Top bar
* [x] Top Sites
* [x] Newsletter
* [x] Sidebar settings
* [x] Extension stats

The main goal of this iteration is to apply a new modular UI with existing logic from the production. The PR does not contain:
* Pinned tabs
* Sponsored links

Notice: the `open preferences` link cannot be implemented, as the browser blocks navigating to this page using every way (anchor, navigation, browser.tabs.create API, etc)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+*.rej
+*.swo
+*.swp
+*.vi
+*~
+
+# OS or Editor folders
 .DS_Store
-src/libs/*
-!.gitkeep
```
Comment (chrmod) [inline .gitignore:5]: this is a rather sensible default

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "root": true,
+  "env": {
+    "es2020": true,
+    "browser": true,
+    "node": true,
```
Comment (chrmod) [inline .eslintrc:6]: no need for node, right?

File: README.md [hunk 3]
```diff
@@ -4,8 +4,8 @@ New Tab page in Ghostery Dawn is implemented as web-extension.
 
 ## Development
 
-Start extension in local Ghostery Dawn installation:
+Start extension:
 
 ```
-npm run start -- --firefox /Applications/Ghostery\ Browser.app/Contents/MacOS/Ghostery
```
Comment (chrmod) [inline README.md:10]: would be nice to keep info about Ghostery Private Browser

File: package.json [hunk 4]
```diff
@@ -23,9 +22,14 @@
...
   "homepage": "https://github.com/ghostery/ghostery-newtab-extension#readme",
   "devDependencies": {
-    "web-ext": "^7.6.0"
+    "patch-package": "^8.0.0",
+    "vite": "^4.4.9",
+    "web-ext": "^7.8.0"
   },
   "dependencies": {
-    "hybrids": "^8.1.14"
+    "eslint": "^8.50.0",
```
Comment (chrmod) [inline package.json]: are those devDependencies?

File: .gitignore (hunk 1 above)
Comment (smalluban) [reply .gitignore:5]: You don't have to create a negative pattern for `gitkeep` to make it save to the git repository. Look for an example at the repo with some gitignore templates: https://github.com/github/gitignore/blob/main/Node.gitignore

File: .eslintrc (hunk 2 above)
Comment (smalluban) [reply .eslintrc:6]: Yes, it's required, for example for `scripts/build.js` - we have node scripts.

File: package.json (hunk 4 above)
Comment (smalluban) [reply package.json]: Right, I'll move them to dev deps.

File: README.md (hunk 3 above)
Comment (smalluban) [reply README.md:10]: I added a new default target as the Ghostery Browser, with the above path.

**Review by chrmod — APPROVED** [review]

**Pull request merged**